Adding glenn2vcf

cmarkello opened the pull request.

cmarkello commented:
For some reason, squashing a commit only creates a new commit stating that I squashed commits.
This is after trying many things in source tree. It seems I can only squash commits on the main branch, but not on the other branches.

And yes I know this won't pass. Just throwing this out here to see if anyone can give me a hint as to what I'm missing.

hannes-ucsc commented:
You seem to squashing while merging. When working on a single-author PR branch, there is never the need for merging. You should be squashing while rebasing (or rebasing interactively) your branch.

hannes-ucsc commented:
> You seem to squashing while merging. 

This should have been: "You seem to combine squashing and merging."

cmarkello committed:
Adding glenn2vcf tool

Fixed versioning of glenn2vcf

Fixed incorrect wrapper destination. Made Ubuntu versions the same between build and runtime Dockerfiles.

Fixed Makefile for glenn2vcf

Added dependencies to runtime Dockerfile

cmarkello commented:
@hannes-ucsc -- ready for review

cmarkello commented:
@fnothaft -- ready for review

jvivian commented:
Jenkins, test this please

cmarkello commented:
Jenkins, test this please

jvivian commented:
LG2M

hannes-ucsc commented:
I think I pointed this out either here or on a different PR: Build and runtime containers have different Ubuntu versions. Am I the only one who thinks this could be a problem? Is this tool statically linked?

cmarkello commented:
Jenkins, test this please

cmarkello commented:
@fnothaft -- ready for review

fnothaft commented on glenn2vcf/Makefile:
```diff
@@ -0,0 +1,32 @@
+build_tool = runtime-container.DONE
+git_commit ?= $(shell git log --pretty=oneline -n 1 -- ../glenn2vcf | cut -f1 -d " ")
+name = quay.io/ucsc_cgl/glenn2vcf
+tag = 2bb4bc2--${git_commit}
```
Nit: prefer long commit hashes for consistency throughout the repository.

fnothaft commented on glenn2vcf/Makefile:
```diff
@@ -0,0 +1,32 @@
+
+${build_output}: build/Dockerfile
+	cd build && docker build -t glenn2vcfbuild .
+	docker run -v ${runtime_fullpath}:/data glenn2vcfbuild cp /home/glenn2vcf/glenn2vcf /data
```
Nit: `s/glenn2vcfbuild/glenn2vcf-build/g` for consistency throughout the repository.

fnothaft commented on glenn2vcf/Makefile:
```diff
@@ -0,0 +1,32 @@
+test: build
+	python test.py
+
+clean:
```
The clean target should remove the image as well.

fnothaft commented:
Generally LGTM, other than a few nits. However, piggybacking off of @hannes-ucsc's question:

> Is this tool statically linked?

You are installing various libraries in the build container (e.g., `libprotoc-dev`) but aren't installing them in the runtime container. Have you verified that the tool works end-to-end using the final container? I'm not sure how things are linked—haven't looked through the glenn2vcf build—but I'm concerned that you could run into runtime object load issues.

cmarkello commented:
Jenkins, test this please

fnothaft commented:
@cmarkello I see this is no longer tagged with `needs work`. Before I merge, what was the answer to my question RE: linking in https://github.com/BD2KGenomics/cgl-docker-lib/pull/133#issuecomment-237084172. If you've run this end-to-end and verified that it does not encounter runtime library load issues, then I will merge.

cmarkello commented:
It does run end to end now as I have added in libraries for the runtime dockerfile. However, I just found out yesterday that glenn2vcf functionality has been merged into the VG call subcommand and it looks like glenn2vcf development will continue in this direction. So right now, there's not really a need to even support this as its own docker container right now.

fnothaft commented:
@cmarkello if there's no need to support this as it's own docker container, then let's close this PR without merging, and remove the glenn2vcf dockerfile? If this sounds good to you, then I'll create an issue.

fnothaft commented:
Er, nevermind RE: creating an issue. I'm going to close this PR for now. If I misunderstood and we still need the glenn2vcf container, please reopen.
